event/listener.php

```diff
@@ -0,0 +1,206 @@
+
+			$post_row['MESSAGE'] = $message;
+			$event['post_row'] = $post_row;
+		}
+	}
+
+	/**
+	 * Show the idea related to the current topic
+	 *
+	 * @param $event
+	 * @return null
+	 * @access public
+	 */
+	public function show_idea($event)
+	{
+		if ($event['forum_id'] != $this->config['ideas_forum_id'])
+		{
+			return;
+		}
+
+		$idea = $this->ideas->get_idea_by_topic_id($event['topic_data']['topic_id']);
+
+		if (!$idea)
+		{
+			return;
+		}
+
+		$mod = $this->auth->acl_get('m_', (int) $this->config['ideas_forum_id']);
+		$own = $idea['idea_author'] === $this->user->data['user_id'];
+
+		if ($mod)
+		{
+			$statuses = $this->ideas->get_statuses();
+			foreach ($statuses as $status)
+			{
+				$this->template->assign_block_vars('statuses', array(
+					'ID'	=> $status['status_id'],
+					'NAME'	=> $this->language->lang($status['status_name']),
+				));
+			}
+		}
+
+		$points = $idea['idea_votes_up'] - $idea['idea_votes_down'];
+		$can_vote = (bool) ($idea['idea_status'] != ideas::STATUS_IMPLEMENTED &&
+			$idea['idea_status'] != ideas::STATUS_DUPLICATE &&
+			$this->auth->acl_get('f_vote', (int) $this->config['ideas_forum_id']) &&
+			$event['topic_data']['topic_status'] != ITEM_LOCKED);
+
+		$this->template->assign_vars(array(
+			'IDEA_ID'			=> $idea['idea_id'],
+			'IDEA_TITLE'		=> $idea['idea_title'],
+			'IDEA_AUTHOR'		=> $this->link_helper->get_user_link($idea['idea_author']),
+			'IDEA_DATE'			=> $this->user->format_date($idea['idea_date']),
+			'IDEA_VOTES'		=> $idea['idea_votes_up'] + $idea['idea_votes_down'],
+			'IDEA_VOTES_UP'		=> $idea['idea_votes_up'],
+			'IDEA_VOTES_DOWN'	=> $idea['idea_votes_down'],
+			'IDEA_POINTS'		=> $this->language->lang('VIEW_VOTES', $points),
+			'IDEA_STATUS'		=> $this->ideas->get_status_from_id($idea['idea_status']),
+			'IDEA_STATUS_LINK'	=> $this->helper->route('phpbb_ideas_list_controller', array('status' => $idea['idea_status'])),
+
+			'IDEA_DUPLICATE'	=> $idea['duplicate_id'],
+			'IDEA_RFC'			=> $idea['rfc_link'],
+			'IDEA_TICKET'		=> $idea['ticket_id'],
+
+			'U_IDEA_DUPLICATE'	=> $this->link_helper->get_idea_link((int) $idea['duplicate_id']),
+
+			'S_IS_MOD'			=> $mod,
+			'S_CAN_EDIT'		=> $mod || $own,
+			'S_CAN_VOTE'		=> $can_vote,
+
+			'U_DELETE_IDEA'		=> $this->link_helper->get_idea_link($idea['idea_id'], 'delete'),
+			'U_CHANGE_STATUS'	=> $this->link_helper->get_idea_link($idea['idea_id'], 'status', true),
+			'U_EDIT_DUPLICATE'	=> $this->link_helper->get_idea_link($idea['idea_id'], 'duplicate', true),
+			'U_EDIT_RFC'		=> $this->link_helper->get_idea_link($idea['idea_id'], 'rfc', true),
+			'U_EDIT_TICKET'		=> $this->link_helper->get_idea_link($idea['idea_id'], 'ticket', true),
+			'U_EDIT_TITLE'		=> $this->link_helper->get_idea_link($idea['idea_id'], 'title', true),
+			'U_REMOVE_VOTE'		=> $this->link_helper->get_idea_link($idea['idea_id'], 'removevote', true),
+			'U_IDEA_VOTE'		=> $this->link_helper->get_idea_link($idea['idea_id'], 'vote', true),
```

Comment:
We cast the ```get_idea_link()``` param to ```int``` above, probably shouldn't or cast it everywhere for consistency.